src/Xhgui/Controller/Run.php

```diff
@@ -9,6 +9,23 @@ public function __construct($app, $profiles, $watches)
         $this->_watches = $watches;
     }
 
+    public function homepage()
+    {
+        // TODO: how to fetch DIC from app?
+        global $di;
```

Comment:
You don't that defeats the purpose of dependency _injection_. Instead you'd change the controller definitions to inject the new dependency.